core/protocols/aws-json-protocol/src/test/java/software/amazon/awssdk/protocols/json/FaultStatusCodeMappingTest.java

```diff
@@ -66,7 +90,17 @@ public static List<TestCase> unmarshal_faultValue_testCases() {
         );
     }
 
-    private static AwsJsonProtocolErrorUnmarshaller makeUnmarshaller(List<ExceptionMetadata> exceptionMetadata) {
+    public static List<QueryErrorTestCase> x_amzn_query_error_testCases() {
+        return Arrays.asList(
+            new QueryErrorTestCase("actualErrorCode;Sender", "actualErrorCode"),
+            new QueryErrorTestCase("actualError CodeSender", "ServiceException"),
+            new QueryErrorTestCase("actualError", "ServiceException"),
+            new QueryErrorTestCase(";Sender", "ServiceException")
```

Comment:
Can we add a test case for empty value as well as null value?

Reply:
sure. will do.